src/core/Akka.TestKit/TestKitBase_Within.cs

```diff
@@ -261,14 +271,32 @@ public abstract partial class TestKitBase
             var prevEnd = _testState.End;
             _testState.End = start + maxDiff;
 
-            T ret;
-            try
-            {
-                ret = await function();
-            }
-            finally
+            T ret = default;
+            using (var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
             {
-                _testState.End = prevEnd;
+                try
+                {
+                    var executionTask = function();
+                    // Limit the execution time block to the maximum allowed execution time.
+                    // 200 milliseconds is added because Task.Delay() timer is not precise and can return prematurely.
+                    var resultTask = await Task.WhenAny(executionTask, Task.Delay(max + 200.Milliseconds(), cts.Token));
```

Comment:
Got it, so this ensures that even if the CTS isn't fired we can still cap the execution time for any spec to `max + 200.milliseconds`